mods/netstimosc.mod

```diff
@@ -0,0 +1,171 @@
+: Modified netstim125 to include oscillating interval between spikes, currently testing
+: ISSUES: - compiles, but cannot use net_move here, gives back "flag error" (commented out)
+: 		  - Study more to see if net_move necessary 
+
+NEURON	{ 
+  ARTIFICIAL_CELL NetStimOsc
+  RANGE number, start, forcestop, freq, status, nspk
+  THREADSAFE : only true if every instance has its own distinct Random
+  POINTER donotuse
+}
+
+PARAMETER {
+	:interval	= 10 (ms) <1e-9,1e9>: time between spikes (msec)
+	number		= 10 <0,1e9>	: number of spikes (independent of noise)
+	start		= 50 (ms)	: start of first spike
+	forcestop 	= 200 (ms)	: stop firing spikes
+	PI 			= 3.14159265358979323846
+	freq 		= 3	: defined externally, but still need to initialize here 
+	status 		= 1 : unused,  ''
+	nspk 		= 10 : unused, ''
+}
+
+ASSIGNED {
+	event (ms)
+	on
+	ispike
+	donotuse
+}
+
+PROCEDURE seed(x) {
+	set_seed(x)
+}
+
+INITIAL {
+	on = 0 : off
+	ispike = 0
+	if (start >= 0 && number > 0) {
+		on = 1
+		: randomize the first spike so on average it occurs at
+		: start + noise*interval
+		event = start + invl(interval(start)) - interval(start)
+		: but not earlier than 0
+		if (event < 0) {
+			event = 0
+		}
+		: no event after time "forcestop"...
+		if (event < forcestop) {
+			net_send(event, 3)
+		}
+	}
+}	
+
+PROCEDURE init_sequence(t(ms)) {
+	if (number > 0) {
+		on = 1
+		event = 0
+		ispike = 0
+	}
+}
+
+COMMENT
+FUNCTION interval (t (ms)) (ms) {
+	interval = 450*sin(2*PI*freq*(t)/1000 + (PI/2))+451
+}
+ENDCOMMENT
+
+: TODO: make the following variables:
+: 	- amplitude (100/freq) controls sparsity between oscillations (should include 'clipping' if statement to enforce upper limit)
```

Comment:
I'll be interested to talk soon about this function. I'm not sure I fully understand everything you are saying here as currently written